packages/yoastseo/src/languageProcessing/helpers/highlighting/getMarkingsInSentence.js

```diff
@@ -143,6 +143,9 @@ function getMarkingsInSentence( sentence, matchesInSentence, matchWordCustomHelp
 			position: {
 				startOffset: token.sourceCodeRange.startOffset,
 				endOffset: token.sourceCodeRange.endOffset,
+				// relative to start of block positions.
+				startOffsetBlock: token.sourceCodeRange.startOffset - ( sentence.parentStartOffset || 0 ),
+				endOffsetBlock: token.sourceCodeRange.endOffset - ( sentence.parentStartOffset || 0 ),
```

Comment:
If I understand this approach correctly, this means that we're assuming `parentStartOffset` is the immediate block. However, sentences can be parts of Paragraph and Heading nodes, in which specifically for paragraph nodes, it can be a child of a nested element right? and this element is not necessarily the block we're interested in